idurar/idurar-erp-crm
# Implementing Error Handling for Backend Server Unavailability in Login/Signup Process

## Description

In this pull request, I have implemented error handling for situations when the backend server is unavailable. Specifically, this addresses the issue where the frontend would indefinitely show a spinner during login or signup attempts without the backend server running. Now, users will receive a clear and meaningful error message indicating the unavailability of the backend, thereby improving user experience and feedback.

## Related Issues

Lack of Error Handling for Backend Unavailability #927

## Steps to Test

1. Start the frontend application.
2. Ensure that the backend server is not running.
3. Attempt to log in or sign up.
4. Observe the new behavior – instead of a continuous spinner, a clear error message should now be displayed, indicating the unavailability of the backend server.

## Screenshots (if applicable)

### Before

![image](https://github.com/idurar/idurar-erp-crm/assets/63133930/8eb3ebe1-1ff3-422c-8c2d-58408744f2f3)

### After
![register](https://github.com/idurar/idurar-erp-crm/assets/63133930/3c466895-e512-4216-aaa8-ecaa75927c14)

![login](https://github.com/idurar/idurar-erp-crm/assets/63133930/e462a02f-d1ed-460e-a9f3-b9dc0fc6fe06)

## Checklist

- [✅] I have tested these changes
- [✅] I have updated the relevant documentation
- [✅] I have commented my code, particularly in hard-to-understand areas
- [✅] I have made corresponding changes to the codebase
- [✅] My changes generate no new warnings or errors
- [✅] The title of my pull request is clear and descriptive
